components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint.util/src/main/java/org/wso2/carbon/identity/mgt/endpoint/util/client/IdentityProviderDataRetrievalClient.java

```diff
@@ -102,6 +111,127 @@ public String getIdPImage(String tenant, String idpName)
         return StringUtils.EMPTY;
     }
 
+    /**
+     * This function retrieves the given configurations of the given federated identity provider using carbon API.
+     * Three HTTP GET calls will be executed as below.
+     * 1. <host>/t/carbon.super/api/server/v1/identity-providers?filter=name+eq+Google
+     * 2. <host>/t/carbon.super/api/server/v1/identity-providers/6719c5cc-5162-44c7-9190-cb74d500f5fc
+     * 3. <host>/t/carbon.super/api/server/v1/identity-providers/6719c5cc-5162-44c7-9190-cb74d500f5fc/
+     * federated-authenticators/R29vZ2xlT0lEQ0F1dGhlbnRpY2F0b3I
+     *
+     * @param tenant     String. The tenant name of the session i.e. carbon.super.
+     * @param idpCode    String. The declared identity provide code i.e. GoogleOIDCAuthenticator.
+     * @param idpName    String. The identity provider name setup at carbon console i.e. Google.
+     * @param configKeys List<String>. The list of keys of required configurations i.e. ClientId, callbackUrl
+     * @return Map<String, String>. Empty String if the configuration is not found, the configuration value otherwise.
+     * @throws IdentityProviderDataRetrievalClientException When there is an error with executing the get calls or
+     *                                                      processing JSON objects.
+     */
+    public Map<String, String> getFederatedIdpConfigs(String tenant, String idpCode, String idpName,
+                                                      List<String> configKeys)
+            throws IdentityProviderDataRetrievalClientException {
+
+        Map<String, String> configMap = new HashMap<>();
+        if (StringUtils.isEmpty(tenant) || StringUtils.isEmpty(idpCode) || StringUtils.isEmpty(idpName) ||
+                configKeys == null
+                || configKeys.isEmpty()) {
+            return configMap;
+        }
+        // i.e. /t/carbon.super/api/server/v1/identity-providers?filter=name+eq+Google
+        JSONObject idpSummaryResult = executePath(tenant, IDP_API_RELATIVE_PATH + IDP_FILTER +
+                Encode.forUriComponent(idpName));
+        if (idpSummaryResult == null) {
+            return configMap;
+        }
+        JSONArray idpSummary = idpSummaryResult.getJSONArray(IDP_KEY);
+        if (idpSummary.length() != 1) {
```

Comment:
Can `idpSummary` be null? In case of an error response from above query?

Reply:
getJSONArray throws JSONException when it is null, thus handled the exception and the response returns and empty map. Fixed with <to be>